vertx-auth-mtls/src/main/java/io/vertx/ext/auth/mtls/CertificateIdentityValidator.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.ext.auth.mtls;
+
+/**
+ * {@link CertificateIdentityValidator} interface can be extended to implement custom certificate identity validator.
+ */
+public interface CertificateIdentityValidator
+{
+    /**
+     * Verifies that identity extracted from certificate is a valid identity.
+     *
+     * @param identity {@code String} string extracted from certificate, uniquely represents client
+     * @return {@code true} if the identity is valid, {@code false otherwise}
+     */
+    boolean isValidIdentity(String identity);
```

Comment:
Prefer throwing exception instead of returning boolean for the validation. The exception conveys better context when it throws. 

```suggestion
    void validateIdentity(String identity) throws InvalidIdentityException;
```